### PR TITLE
Implementation improve

### PR DESCRIPTION
**Pull request by arfenis**
- Create follow workflow to allow the user to use Social to follow entities.
- Delete unused code.
- Create follows tests.
- Improve TotalInteractions Traits

### CODE REVIEW

File: src/Social/Contract/Interactions/InteractionsTrait.php [hunk 1]
```diff
@@ -19,9 +19,9 @@ trait InteractionsTrait
      * @param string $action
      * @return void
      */
-    public function interact(string $action): void
+    public function interact(string $action, $data): void
```
Comment (kaioken) [inline src/Social/Contract/Interactions/InteractionsTrait.php]: @arfenis can we know what data type $data is?

File: src/Social/Listener/Users.php [hunk 2]
```diff
@@ -70,6 +71,9 @@ public function reply(Event $event, ModelInterface $entity): void
      */
     public function follow(Event $event, ModelInterface $entity): void
     {
-        Interactions::add(Di::getDefault()->get('userData'), $entity, InteractionsModel::FOLLOWING);
+        $user = Di::getDefault()->get('userData');
```
Comment (kaioken) [inline src/Social/Listener/Users.php:74]: can we use getting the userData from baka directly ?
https://github.com/bakaphp/baka/blob/0.6/src/Auth/UserProvider.php#L32

File: src/Social/Services/Follow.php [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    /**
+     * Follow and unfollow an entity if its exist.
+     *
+     * @param UserInterface $userFollowing User that is following
+     * @param ModelInterface $entity Entity that is being followed
+     * @return boolean
+     */
+    public static function userFollow(UserInterface $userFollowing, ModelInterface $entity): bool
+    {
+        $follow = UsersFollows::findFirst([
```
Comment (kaioken) [inline src/Social/Services/Follow.php:44]: @arfenis can we use findOrCreate?

**Review by kaioken — DISMISSED** [review]

File: src/Social/Contract/Interactions/InteractionsTrait.php (hunk 1 above)
Comment (arfenis) [reply src/Social/Contract/Interactions/InteractionsTrait.php]: Ups, forgot.

**Review by kaioken — APPROVED** [review]

**Pull request merged**